### ISSUE TITLE
Uncaught TypeError: Path must be a string. Received undefined

### ISSUE THREAD

**Issue by bajo** [artivis/build-catkin-tools#1]
[Enter steps to reproduce below:]

1. Open a file within a catkin package (inside one of my catkin workspaces)
2. Try to compile (catkin build) from within Atom by pressing F9.

It seems that the package path of the currently edited file is not passed to constructor(source_dir) on line 29 in cmake.js. 

Any hints how to fix this? 
Thanks

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Ubuntu 14.04.5
**Thrown From**: [build-catkin-tools](https://github.com/artivis/build-catkin-tools) package, v0.3.2

### Stack Trace

Uncaught TypeError: Path must be a string. Received undefined

```
At path.js:7

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.join (path.js:1213:7)
    at CatkinBuildProvider.isCatkinToolsWorkspace (/home/bajo/.atom/packages/build-catkin-tools/lib/cmake.js:71:33)
    at CatkinBuildProvider.isEligible (/home/bajo/.atom/packages/build-catkin-tools/lib/cmake.js:66:19)
    at /home/bajo/.atom/packages/build/lib/target-manager.js:68:30
    at Array.filter (native)
    at /home/bajo/.atom/packages/build/lib/target-manager.js:68:10
    at Array.map (native)
    at TargetManager.refreshTargets (/home/bajo/.atom/packages/build/lib/target-manager.js:58:39)
    at TargetManager.getTargets (/home/bajo/.atom/packages/build/lib/target-manager.js:216:19)
    at Object.startNewBuild (/home/bajo/.atom/packages/build/lib/build.js:111:40)
    at /home/bajo/.atom/packages/build/lib/build.js:270:14
    at saveAndContinue (/home/bajo/.atom/packages/build/lib/build.js:277:7)
    at Object.doSaveConfirm (/home/bajo/.atom/packages/build/lib/build.js:281:7)
    at Object.build (/home/bajo/.atom/packages/build/lib/build.js:264:10)
    at atom-workspace.<anonymous> (/home/bajo/.atom/packages/build/lib/build.js:25:69)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:259:29)
    at CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:3:59)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/usr/share/atom/resources/app.asar/src/command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchApplicationMenuCommand (/usr/share/atom/resources/app.asar/src/atom-environment.js:1175:28)
    at EventEmitter.outerCallback (/usr/share/atom/resources/app.asar/src/application-delegate.js:319:25)
    at emitThree (events.js:116:13)
    at EventEmitter.emit (events.js:194:7)
```

### Commands

```
     -0:21.2.0 build:toggle-panel (input.hidden-input)
  2x -0:16.5.0 build:trigger (input.hidden-input)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "clang-format"
    ],
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db",
      ".pioenvs",
      ".piolibdeps",
      ".clang_complete",
      ".gcc-flags.json"
    ],
    "telemetryConsent": "limited"
  },
  "build-catkin-tools": {
    "ros_version": "indigo"
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.16 (inactive)
autocomplete-clang, v0.10.0 (active)
build, v0.67.0 (active)
build-catkin-tools, v0.3.2 (active)
busy, v0.7.0 (active)
clang-format, v1.25.0 (inactive)
file-icons, v2.0.8 (active)
language-ini, v1.17.0 (active)
linter, v1.11.21 (active)
linter-gcc, v0.6.15 (active)
minimap, v4.26.6 (active)
platformio-ide, v1.7.1 (active)
platformio-ide-terminal, v2.2.2 (active)
tool-bar, v1.0.1 (active)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (active)
one-light-ui, v1.8.2 (inactive)
one-dark-syntax, v1.6.0 (active)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.14.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.33.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.43.2 (active)
bracket-matcher, v0.85.1 (active)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.204.5 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.2.0 (active)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.5 (active)
status-bar, v1.6.0 (active)
styleguide, v0.48.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.39.0 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.1 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by artivis** [comment]
Hi,
thanks for reporting. I will investigate it asap.

**Comment by bajo** [comment]
Hi. 

Great, thanks a lot.

**Comment by artivis** [comment]
Fixed in #2 .
